### PR TITLE
Replace ClipboardJS with copyToClipboard

### LINKED ISSUE #18788: Copy token button no longer works
When creating a token, a copy button is provided for convenience. This no longer copies the token.

image

## Expected Behavior

Token is copied

## Current Behavior

token is not copied

## Steps to Reproduce (for bugs)

1. Create a new token
2. Click on the copy button

## Context

Testing sidecars in 6.0.beta.3

There are no web browser console errors.

## Your Environment

* Graylog Version: 6.0.beta.3
* Mac OS Sanoma 14.4.1
* Google Chrome Version 123.0.6312.87 (Official Build) (arm64)
* Also tested with Windows 11 and Firefox (same issue)

### PR DESCRIPTION
**Pull request by ousmaneo**
For the copy to clipboard in the scratchpad, we were using ClipboardJS, which has stopped working somehow.
These changes replace the usage of ClipboardJS with `copyToClipboard` function already used for the keyboard shortcut.

Partially fix #18788
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/scratchpad/ScratchpadModal.tsx [hunk 1]
```diff
@@ -251,10 +233,7 @@ const Scratchpad = () => {
           </StatusMessage>
 
           <ButtonGroup>
-            <Button data-clipboard-button
-                    data-clipboard-target={`#${TEXTAREA_ID}`}
-                    id="scratchpad-actions"
-                    title="Copy">
+            <Button id="scratchpad-actions" onClick={() => copyCallback()} title="Copy">
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/scratchpad/ScratchpadModal.tsx:236]: We can just pass the function itself, that avoids creating an anonymous function on each render:

```suggestion
            <Button id="scratchpad-actions" onClick={copyCallback} title="Copy">
```

File: graylog2-web-interface/yarn.lock [hunk 2]
```diff
@@ -5193,7 +5193,7 @@ clean-css@^5.1.5:
   dependencies:
     source-map "~0.6.0"
 
-clipboard@*, clipboard@^2.0.0:
+clipboard@*:
```
Comment (dennisoelkers) [inline graylog2-web-interface/yarn.lock:5196]: Do you know why there is still an entry in the lockfile after the dependency has been removed?

File: graylog2-web-interface/src/theme/docs/Colors.jsx [hunk 3]
```diff
@@ -99,9 +86,7 @@ const ColorSwatch = ({ className, color, name, copyText }) => {
           Copied!
         </StyledTooltip>
         <Swatch color={color}
-                data-clipboard-button
-                data-clipboard-text={copyText}
-                ref={swatchRef}>
+                onClick={() => copyCallback()}>
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/theme/docs/Colors.jsx:89]: Same here:
```suggestion
                onClick={copyCallback}>
```

File: graylog2-web-interface/src/theme/docs/Colors.jsx [hunk 4]
```diff
@@ -65,29 +65,16 @@ const Swatch = styled.button(({ color, theme }) => css`
 
 const ColorSwatch = ({ className, color, name, copyText }) => {
   const [opened, setOpened] = useState(false);
-  const swatchRef = useRef();
 
-  useEffect(() => {
-    let clipboard;
+  const copyCallback = () => {
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/theme/docs/Colors.jsx:69]: Can we wrap this with `useCallback` to keep the reference stable?

**Review by dennisoelkers — COMMENTED** [review]
Looks good, great to see `clipboard` go. Just added some remarks and one question inline.

File: graylog2-web-interface/yarn.lock (hunk 2 above)
Comment (ousmaneo) [reply graylog2-web-interface/yarn.lock]: There was still a reference to ` @types/clipboard` that was causing this; I removed it, too.

**Review by dennisoelkers — APPROVED** [review]
Great change! Tested and works fine. Thanks! :pray:

**Comment by ousmaneo** [comment]
Thank you for the review !

**Pull request merged**